KinectingPepper/UI/RewindPage.xaml.cs

```diff
@@ -36,20 +44,113 @@ private void cbxCameraType_SelectionChanged(object sender, SelectionChangedEvent
 
         private void selectFile_Click(object sender, RoutedEventArgs e)
         {
-            
+            if (_playBackFrames)
+            {
+                _playBackFrames = false;
+                CompositionTarget.Rendering -= CompositionTarget_Rendering;
+                _currentFrameNumber = 0;
+                _timeLastFrameRender = DateTime.MinValue;
+                _framesFromDisk = null;
+            }
+
+            OpenFileDialog openFileDialog1 = new OpenFileDialog();
+
+            openFileDialog1.InitialDirectory = @"C:\images\Pepper\";
+            openFileDialog1.Filter = "XML files (*.xml)|*.xml";
+
+            if (openFileDialog1.ShowDialog() == true)
+            {
+                try
+                {
+                    _framesFromDisk = PersistFrames.Instance.DeserializeFromXML(openFileDialog1.FileName);
+                    _playBackFrames = true;
+
+                    slrFrameProgress.Maximum = 0;
+                    slrFrameProgress.Maximum = _framesFromDisk.Count - 1;
+
+                    CompositionTarget.Rendering += CompositionTarget_Rendering;
+                }
+                catch (Exception ex)
```

Comment:
Voor het mooie alleen de Exceptions afvangen die gegooid kunnen worden. Anders kan je ook random exceptions afvangen wat waarschijnlijk niet de bedoeling is.